BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsClient.kt

```diff
@@ -145,60 +147,68 @@ internal class AnalyticsClient @VisibleForTesting constructor(
-                    analyticsUrl,
-                    analyticsRequest.toString(),
-                    null,
-                    authorization,
-                    HttpNoResponse()
-                )
-            }
+            httpClient.post(
+                FPTI_ANALYTICS_URL,
+                analyticsRequest.toString(),
+                null,
+                authorization,
+                HttpNoResponse()
+            )
         } catch (e: JSONException) { /* ignored */
         }
     }
 
     @Throws(JSONException::class)
     private fun serializeEvents(
-        authorization: Authorization?, events: List<AnalyticsEvent>, metadata: DeviceMetadata
+        authorization: Authorization?,
+        events: List<AnalyticsEvent>,
+        metadata: DeviceMetadata
     ): JSONObject {
-        val requestObject = JSONObject()
+        val batchParamsJSON = metadata.toJSON()
         authorization?.let {
             if (it is ClientToken) {
-                requestObject.put(AUTHORIZATION_FINGERPRINT_KEY, it.bearer)
+                batchParamsJSON.put(AUTHORIZATION_FINGERPRINT_KEY, it.bearer)
             } else {
-                requestObject.put(TOKENIZATION_KEY, it.bearer)
+                batchParamsJSON.put(TOKENIZATION_KEY, it.bearer)
             }
         }
 
-        requestObject.put(META_KEY, metadata.toJSON())
-        val eventObjects = JSONArray()
-        var eventObject: JSONObject
+        val eventsContainerJSON = JSONObject()
+        eventsContainerJSON.put(BATCH_PARAMS_KEY, batchParamsJSON)
+
+        val eventParamsJSON = JSONArray()
         for (analyticsEvent in events) {
-            eventObject = JSONObject()
-                .put(KIND_KEY, analyticsEvent.name)
-                .put(PAYPAL_CONTEXT_ID_KEY, analyticsEvent.payPalContextId)
+            val singleEventJSON = JSONObject()
+                .put(EVENT_NAME_KEY, analyticsEvent.name)
+                .putOpt(PAYPAL_CONTEXT_ID_KEY, analyticsEvent.payPalContextId)
                 .put(TIMESTAMP_KEY, analyticsEvent.timestamp)
-            eventObjects.put(eventObject)
+                .put(TENANT_NAME_KEY, "Braintree")
+            eventParamsJSON.put(singleEventJSON)
         }
-        requestObject.put(ANALYTICS_KEY, eventObjects)
-        return requestObject
+        eventsContainerJSON.put(EVENT_PARAMS_KEY, eventParamsJSON)
+
+        // Single-element "events" array required by FPTI formatting
+        val eventsArray = JSONArray(arrayOf(eventsContainerJSON))
+        return JSONObject().put(EVENTS_CONTAINER_KEY, eventsArray)
     }
 
     companion object {
-        private const val ANALYTICS_KEY = "analytics"
-        private const val PAYPAL_CONTEXT_ID_KEY = "payPalContextId"
-        private const val KIND_KEY = "kind"
-        private const val TIMESTAMP_KEY = "timestamp"
-        private const val META_KEY = "_meta"
+        private const val FPTI_ANALYTICS_URL = "https://api-m.paypal.com/v1/tracking/batch/events"
```

Comment:
Do we need to have a separate endpoint for sandbox? Or can we filter the data by environment?

Reply:
we can filter data by environment 👍